app/services/xccdf_report_migration.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+# Migrates ref IDs from reports to comply with the expected standard.
+# Profile ref IDs should look like "xccdf_org.ssgproject.content_profile_*"
+# Rule ref IDs should look like "xccdf_org.ssgproject.content_rule_*"
+class XCCDFReportMigration
+  def initialize(account, noop)
+    @account = account
+    @noop = noop
+  end
+
+  def logger
+    Rails.logger
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def run
+    xccdf_profiles = broken_profiles
+    logger.info "Found #{xccdf_profiles.count} profiles from XCCDF reports."\
+      ' Renaming profiles...'
+    rename_profiles(xccdf_profiles)
+    logger.info "Profiles rename finished \n-------------------------------\n"\
+      'Some profiles might have not been renamed if the name '\
+      'conflicts with another profile in the same account.'\
+      "Let's proceed to migrating the XCCDF profile results into the"\
+      ' DS profile results'
+    xccdf_profiles = broken_profiles
+    logger.info "Found #{xccdf_profiles.count} profiles from XCCDF reports."
+    migrate_profiles(xccdf_profiles)
+  end
+  # rubocop:enable Metrics/MethodLength
+
+  def broken_profiles
+    Profile.where(account_id: @account.id).where.not(
+      'ref_id LIKE ?', 'xccdf_org.ssgproject.content_profile_%'
+    )
+  end
+
+  def rename_profiles(xccdf_profiles)
+    return if @noop
+
+    xccdf_profiles.map do |profile|
+      profile.update(
+        ref_id: "xccdf_org.ssgproject.content_profile_#{profile.ref_id}"
+      )
+      migrate_rules(profile.rules, profile) if profile.errors.empty?
+    end
+  end
+
+  def find_matching_profile(xccdf_profile)
+    Profile.where(
+      account_id: @account.id,
+      ref_id: "xccdf_org.ssgproject.content_profile_#{xccdf_profile.ref_id}"
+    ).first
+  end
+
+  def migrate_profile_hosts(xccdf_profile, ds_profile)
+    return if @noop
+
+    logger.info(ProfileHost
+      .where(profile_id: xccdf_profile.id)
+      .map { |profile_host| profile_host.update(profile_id: ds_profile.id) })
```

Comment:
This runs validations - it's probably fine as every Host doesn't have more than a couple profiles usually and this task runs once. It helps us avoid the issue with `update_all` triggering DB validations and therefore an exception that kills the job.